### PR TITLE
Move proxy init to service unit file ExecStartPost

### PR DESCRIPTION
**Pull request by tanabarr**
When `scanner-proxy` starts, a data event needs to be triggered on the socket in order for it to be detected by the service listener and forwarded to the `device-aggregator`. This initialisation needs to occur after the service starts.

Signed-off-by: Tom Nabarro <tom.nabarro@outlook.com>

### CODE REVIEW

File: IML/ScannerProxyDaemon/systemd-units/scanner-proxy.service [hunk 1]
```diff
@@ -6,6 +6,10 @@ After=device-scanner.service
 [Service]
 Restart=always
 Environment=NODE_ENV=production
+# allow agent to initialise settings directory
+ExecStartPre=/bin/sleep 5
```
Comment (jgrund) [inline IML/ScannerProxyDaemon/systemd-units/scanner-proxy.service]: We should not need to sleep for this.

File: IML/ScannerProxyDaemon/systemd-units/scanner-proxy.service [hunk 2]
```diff
@@ -6,6 +6,10 @@ After=device-scanner.service
...
 ExecStart=/usr/bin/node /usr/lib64/iml-scanner-proxy-daemon/scanner-proxy-daemon
+# allow daemon to connect to socket before triggering data event
+ExecStartPost=/bin/bash -c '/usr/bin/sleep 2; /usr/bin/echo \'"Info"\' | /usr/bin/socat - UNIX-CONNECT:/var/run/device-scanner.sock'
```
Comment (jgrund) [inline IML/ScannerProxyDaemon/systemd-units/scanner-proxy.service]: Shouldn't need to do this either.

**Comment by tanabarr** [comment]
yes, the sleeps are just a workaround to get a working prototype to progress with while we find solutions to the underlying issues

**Comment by jgrund** [comment]
This is superseded by #10

**Comment by tanabarr** [comment]
this isn't completely covered by #10 , once installed the aggregator won't be updated until the device-scanner updates its output

**Pull request closed without merging**